client/packages/common/src/ui/components/navigation/ListOptions/ListOptions.tsx

```diff
@@ -0,0 +1,81 @@
+import { CheckIcon, ChevronDownIcon } from '@common/icons';
+import {
+  List,
+  ListItemIcon,
+  ListItem,
+  ListItemText,
+  Divider,
+  Box,
+} from '@mui/material';
+import React from 'react';
+
+export type ListOptionValues = {
+  id: string;
+  value: string;
+};
+
+interface ListProps {
+  onClick: (id: string) => void;
+  options: ListOptionValues[];
+  currentId?: string;
+  enteredLineIds?: string[];
+}
+
+export const ListOptions = ({
+  onClick,
+  options,
+  currentId,
+  enteredLineIds,
+}: ListProps) => {
+  const startIcon = (
+    <CheckIcon
+      style={{
+        backgroundColor: '#33A901',
+        borderRadius: '50%',
+        padding: '2px',
+        color: 'white',
+        height: 18,
+        width: 18,
+      }}
+    />
+  );
+
+  const endIcon = (
+    <ChevronDownIcon
+      style={{ width: 17, height: 17, transform: 'rotate(-90deg)' }}
+    />
+  );
+
+  return (
+    <List sx={{ padding: 0 }}>
+      {options?.map((option, _) => (
+        <React.Fragment key={option.id}>
+          <ListItem
+            sx={{ padding: '5px 0px' }}
+            onClick={() => onClick(option.id)}
+          >
+            <ListItemIcon sx={{ padding: 0, minWidth: 25 }}>
+              {enteredLineIds?.includes(option.id) ? (
+                startIcon
+              ) : (
+                <Box style={{ visibility: 'hidden' }}>{startIcon}</Box>
+              )}
```

Comment:
This one is quite funky - behind the scenes it appears to still be rendering the icon it's just invisible. But if you remove the icon JSX from here it renders the same anyway, just shadow dom doing less work?